qiskit/circuit/library/logical_expression_oracle.py

```diff
@@ -0,0 +1,175 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A quantum oracle constructed from a DIMACS format logical expression."""
+import re
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+
+
+class LogicalExpressionOracle(QuantumCircuit):
```

Comment:
Shouldn't `LogicalExpressionOracle` be an `Instruction`?

Reply:
It should be a circuit, since it's going in the circuit library 🙂 also then it's directly compatible with the `GroverOperator` which takes a circuit as oracle